dimod/bqm/utils.pyx

```diff
@@ -40,6 +42,78 @@ cdef object as_numpy_scalar(double a, np.dtype dtype):
     """Note that the memory is interpreted to match dtype, not a cast"""
     return PyArray_Scalar(&a, dtype, None)
 
+def cylmin(cyBQM bqm, **kwargs):
+    if bqm.num_variables == 0:
+        if 'default' in kwargs:
+            return kwargs['default']
+        else:
+            raise ValueError("Argument is an empty sequence")
+
+    cdef double min_ = sys.float_info.max
+    cdef Py_ssize_t vi
+    for vi in range(bqm.bqm_.num_variables()):
+        val = bqm.bqm_.get_linear(vi)
+        if val < min_:
+            min_ = val
+    
+    return min_
+
+def cylmax(cyBQM bqm, **kwargs):
+    if bqm.num_variables == 0:
+        if 'default' in kwargs:
+            return kwargs['default']
+        else:
+            raise ValueError("Argument is an empty sequence")
+
+    cdef double max_ = -sys.float_info.max
+    cdef Py_ssize_t vi
+    for vi in range(bqm.bqm_.num_variables()):
+        val = bqm.bqm_.get_linear(vi)
+        if val > max_:
+            max_ = val
+    
+    return max_
+
+def cyqmin(cyBQM bqm, **kwargs):
+    if bqm.num_interactions == 0:
+        if 'default' in kwargs:
+            return kwargs['default']
+        else:
+            raise ValueError("Argument is an empty sequence")
+
+    cdef double min_ = sys.float_info.max
+    cdef Py_ssize_t vi
+    for vi in range(bqm.bqm_.num_variables()):
+        span = bqm.bqm_.neighborhood(vi)
+
+        while span.first != span.second:
+            if deref(span.first).second < min_:
+                min_ = deref(span.first).second
+
+            inc(span.first)
+
+    return min_
+
+def cyqmax(cyBQM bqm, **kwargs):
+    if bqm.num_interactions == 0:
+        if 'default' in kwargs:
+            return kwargs['default']
+        else:
+            raise ValueError("Argument is an empty sequence")
+
+    cdef double max_ = -sys.float_info.max
+
+    cdef Py_ssize_t vi
+    for vi in range(bqm.bqm_.num_variables()):
+        span = bqm.bqm_.neighborhood(vi)
+        
+        while span.first != span.second:
```

Comment:
As a detail, the biases are kept symmetrically, so you could do 
```
while span.first != span.second and deref(span.first).first < vi:
```
which would speed this up a bit.